lib/semantics/tools.cc

```diff
@@ -399,6 +430,22 @@ const Symbol *HasEventOrLockPotentialComponent(
   return nullptr;
 }
 
+bool IsOrContainsEventOrLockComponent(const Symbol &symbol) {
+  if (const Symbol * root{GetAssociationRoot(symbol)}) {
+    if (const auto *details{root->detailsIf<ObjectEntityDetails>()}) {
+      if (const DeclTypeSpec * type{details->type()}) {
+        if (const DerivedTypeSpec * derived{type->AsDerived()}) {
+          if (IsEventTypeOrLockType(derived) ||
```

Comment:
You could have `return IsEventTypeOrLockType(derived) || HasEventOrLockPotentialComponent(*derived);`
Since we're using this form in other places too.

Reply:
Good catch!  I'll change this.